src/cdk/datetime/datetime.md

```diff
@@ -0,0 +1,94 @@
+The `datetime` package offers building blocks for custom date and time-based components.
+
+### DateAdapter
+The `DateAdapter` provides a uniform way to deal with different date representations. The CDK `DateAdapter` facilitates
+use of any date data structure for date and time-based components, such as the material datepicker, and gives a
+foundation for more concrete material `DateAdapter` implementations.
```

Comment:
I would restructure the date adapter section like this:
```markdown
### Using a DateAdapter
A `DateAdapter` provides a uniform way to deal with different date representations. Whenever
a directive needs to read any date information, it gets that information from the `DateAdapter`.
The adapter then reads the information using the appropriate API from the underlying
date implementation.

#### Pre-made adapters

##### Native JavaScript Dates
The CDK includes one default `DateAdapter` implementation: `NativeDateAdapter`. This adapter
uses the native JavaScript `Date` object and native APIs in the `Intl` namespace. The native
`Date` is used as the default implementation because it has no additional dependencies;
choosing a full date/time localization library is best left to application-developers,
so the CDK does not include or force any specific solution.

**Please note**: JavaScript's native date-parsing abilities are extremely limited.
Parsing of date strings with the `Date` constructor and `Date.parse` is strongly
discouraged due to browser differences and inconsistencies. Support for RFC 2822
format strings is by convention only. Support for ISO 8601 formats differs in that date-only
strings (e.g. "1970-01-01") are treated as UTC, not local. It is strongly recommended that
developers use a full date/time localization library to support a variety of locales and for
browser consistency.

##### MomentJS
An adapter for [Moment.js](https://momentjs.com), `MomentDateAdapter`,
is available from the npm package `@angular/material-moment-adapter`. 

...

#### Building a custom adapter
...
```